flang/lib/Evaluate/intrinsics.cpp

```diff
@@ -2333,6 +2344,14 @@ std::optional<SpecificCall> IntrinsicInterface::Match(
       CHECK(FloatingType.test(*category));
       resultType = DynamicType{*category, defaults.doublePrecisionKind()};
       break;
+    case KindCode::quadPrecision:
+      CHECK(result.categorySet == CategorySet{*category});
+      CHECK(FloatingType.test(*category));
+      resultType = DynamicType{*category, defaults.quadPrecisionKind()};
+#if !(defined(FLANG_RUNTIME_F128_MATH_LIB) || HAS_LDBL128)
```

Comment:
Don't use configuration macros for this test.  Use the TargetCharacteristics structure instead.